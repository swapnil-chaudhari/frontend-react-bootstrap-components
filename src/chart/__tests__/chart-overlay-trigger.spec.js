import React from 'react';
import renderShallow from 'render-shallow';
import noop from 'src/utils/noop';
import ChartOverlayTrigger from '../chart-overlay-trigger';

describe('<ChartOverlayTrigger>', () => {
    describe('when it renders', () => {
        const onMouseMove = noop;
        const onMouseLeave = noop;
        const onClick = noop;

        let component;

        beforeAll(() => {
            component = renderShallow(
                <ChartOverlayTrigger
                    className="extra"
                    x={ 0 }
                    y={ 0 }
                    width={ 100 }
                    height={ 200 }
                    onMouseMove={ onMouseMove }
                    onMouseLeave={ onMouseLeave }
                    onClick={ onClick }
                />
            ).output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
