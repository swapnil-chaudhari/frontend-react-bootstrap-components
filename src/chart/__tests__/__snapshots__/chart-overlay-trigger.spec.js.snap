// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartOverlayTrigger> when it renders has proper markup 1`] = `
<g>
  <rect
    className="ChartOverlayTrigger extra"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "height": 200,
        "left": 0,
        "top": 0,
        "width": 100,
      }
    }
  />
</g>
`;
